File: app/src/main/java/com/otoniel/alura/ui/activity/ListStudentsActivity.java
package com.otoniel.alura.ui.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.ContextMenu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.otoniel.alura.R;
import com.otoniel.alura.model.Student;
import com.otoniel.alura.ui.view.StudentListView;

import static com.otoniel.alura.ui.activity.ConstantsActivities.KEY_STUDENT_EDIT;

public class ListStudentsActivity extends AppCompatActivity implements AdapterView.OnItemClickListener {

    final private StudentListView studentListView = new StudentListView(this);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list_students);
        setTitle(getResources().getString(R.string.acitivy_list_student_button));

        createListView();
        handleFabButton();
    }

    @Override
    public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMenuInfo menuInfo) {
        super.onCreateContextMenu(menu, v, menuInfo);
        getMenuInflater().inflate(R.menu.activity_student_list_menu, menu);
    }

    @Override
    public boolean onContextItemSelected(@NonNull MenuItem item) {
        final int itemMenuId = item.getItemId();
        AdapterView.AdapterContextMenuInfo menuInfo = (AdapterView.AdapterContextMenuInfo) item.getMenuInfo();
        Student student = studentListView.getStudent(menuInfo.position);
        if (itemMenuId == R.id.activity_student_list_menu_remove) {
            studentListView.showDialogRemove(student);
        } else if (itemMenuId == R.id.activity_student_list_menu_edit) {
            initFormActivityEdit(student);
        }
        return super.onContextItemSelected(item);
    }

    private void handleFabButton() {
        FloatingActionButton actionButton = findViewById(R.id.list_student_activity_fab_add);
        actionButton.setOnClickListener(v -> initFormActivity());
    }

    private void initFormActivity() {
        startActivity(
                new Intent(this, FormStudentActivity.class)
        );
    }

    private void initFormActivityEdit(Student student) {
        Intent formActivity = new Intent(this, FormStudentActivity.class);
        formActivity.putExtra(KEY_STUDENT_EDIT, student);
        startActivity(formActivity);
    }

    @Override
    protected void onResume() {
        super.onResume();
        studentListView.refreshAdapter();
    }

    private void createListView() {
        ListView listView = findViewById(R.id.activity_list_student_listview);
        listView.setAdapter(studentListView.getAdapterView());
        listView.setOnItemClickListener(this);
        registerForContextMenu(listView);
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        Student student = (Student) parent.getItemAtPosition(position);
        initFormActivityEdit(student);
    }
}

File: app/src/main/java/com/otoniel/alura/ui/activity/ListStudentsActivity.kt
package com.otoniel.alura.ui.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.otoniel.alura.R
import com.otoniel.alura.model.Student
import android.widget.AdapterView.OnItemClickListener
import com.otoniel.alura.ui.view.StudentListView
import android.view.ContextMenu
import android.view.ContextMenu.ContextMenuInfo
import android.widget.AdapterView.AdapterContextMenuInfo
import com.google.android.material.floatingactionbutton.FloatingActionButton
import android.content.Intent
import android.view.MenuItem
import android.view.View
import android.widget.AdapterView
import android.widget.ListView

class ListStudentsActivity : AppCompatActivity(), OnItemClickListener {
    private lateinit var studentListView: StudentListView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_list_students)
        title = resources.getString(R.string.acitivy_list_student_button)

        studentListView = StudentListView(this@ListStudentsActivity)

        createListView()
        handleFabButton()
    }

    override fun onCreateContextMenu(menu: ContextMenu, v: View, menuInfo: ContextMenuInfo) {
        super.onCreateContextMenu(menu, v, menuInfo)
        menuInflater.inflate(R.menu.activity_student_list_menu, menu)
    }

    override fun onContextItemSelected(item: MenuItem): Boolean {
        val itemMenuId = item.itemId
        val menuInfo = item.menuInfo as AdapterContextMenuInfo
        val student = studentListView.getStudent(menuInfo.position)

        if (itemMenuId == R.id.activity_student_list_menu_remove) {
            studentListView.showDialogRemove(student)

        } else if (itemMenuId == R.id.activity_student_list_menu_edit) {
            initFormActivityEdit(student)
        }
        return super.onContextItemSelected(item)
    }

    private fun handleFabButton() {
        val actionButton = findViewById<FloatingActionButton>(R.id.list_student_activity_fab_add)
        actionButton.setOnClickListener { v: View? -> initFormActivity() }
    }

    private fun initFormActivity() {
        startActivity(
            Intent(this, FormStudentActivity::class.java)
        )
    }

    private fun initFormActivityEdit(student: Student) {
        val formActivity = Intent(this, FormStudentActivity::class.java)
        formActivity.putExtra(ConstantsActivities.Companion.KEY_STUDENT_EDIT, student)
        startActivity(formActivity)
    }

    override fun onResume() {
        super.onResume()
        studentListView.refreshAdapter()
    }

    private fun createListView() {
        val listView = findViewById<ListView>(R.id.activity_list_student_listview)
        listView.adapter = studentListView.adapterView
        listView.onItemClickListener = this
        registerForContextMenu(listView)
    }

    override fun onItemClick(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        val student = parent.getItemAtPosition(position) as Student
        initFormActivityEdit(student)
    }
}